docs/benchmarks/benchmarking.md

```diff
@@ -0,0 +1,254 @@
+# limitations under the License. -->
+
+# Dynamo Benchmarking Guide
+
+This benchmarking framework lets you compare performance across any combination of:
+- **DynamoGraphDeployments** (automatically deployed from your manifests)
+- **External HTTP endpoints** (existing services, vLLM, TensorRT-LLM, etc.)
+
+You can mix and match these in a single benchmark run using custom labels. Configure your DynamoGraphDeployment manifests for your specific models, hardware, and parallelization needs.
+
+## What This Tool Does
+
+The framework is a wrapper around `genai-perf` that:
+- Deploys user-specified `DynamoGraphDeployments` automatically
+- Benchmarks any HTTP endpoints (no deployment needed)
+- Runs concurrency sweeps across configurable load levels
+- Generates comparison plots with your custom labels
+- Works with any HuggingFace-compatible model on NVIDIA GPUs (H200, H100, A100, etc.)
+- Runs locally and connects to your Kubernetes deployments/endpoints
+
+**Default sequence lengths**: Input: 2000 tokens, Output: 256 tokens (configurable with `--isl` and `--osl`)
+
+**Important**: The `--model` parameter configures GenAI-Perf for benchmarking and provides logging context. The actual model loaded is determined by your deployment manifests. Only one model can be benchmarked at a time across all inputs to ensure fair comparison. The default `--model` value in the benchmarking script is `deepseek-ai/DeepSeek-R1-Distill-Llama-8B`, but it must match the model in the manifest(s) and the model deployed at the endpoint(s).
+
+## Prerequisites
+
+1. **Kubernetes cluster with NVIDIA GPUs and Dynamo namespace setup** - You need a Kubernetes cluster with eligible NVIDIA GPUs and a properly configured namespace for Dynamo benchmarking. See the [deploy/utils/README](../../deploy/utils/README.md) for complete setup instructions.
+
+2. **kubectl access** - You need `kubectl` installed and configured to access your Kubernetes cluster. All other required tools (GenAI-Perf, Python, etc.) are included in the Dynamo containers. If you are not working within a Dynamo container, you can install the necessary requirements using `deploy/utils/requirements.txt`. *Note: if you are on Ubuntu 22.04 or lower, you will also need to build perf_analyzer [from source](https://github.com/triton-inference-server/perf_analyzer/blob/main/docs/install.md#build-from-source).*
+
+## Quick Start Examples
+
+```bash
+export NAMESPACE=benchmarking
+
+# Compare multiple DynamoGraphDeployments
+./benchmarks/benchmark.sh --namespace $NAMESPACE \
+   --input agg=components/backends/vllm/deploy/agg.yaml \
+   --input disagg=components/backends/vllm/deploy/disagg.yaml
+
+# Compare Dynamo vs external endpoint
+./benchmarks/benchmark.sh --namespace $NAMESPACE \
+   --input dynamo=components/backends/vllm/deploy/disagg.yaml \
+   --input vllm-baseline=http://localhost:8000
+
+# Compare three different configurations
+./benchmarks/benchmark.sh --namespace $NAMESPACE \
```

Comment:
basic question: the terms "dynamo vs external" seem a little odd here as well as "no deployment needed"

does everything go through a deployment - I would assume you'd want to compare things in the same cluster - do we need to include the "non dynamo" path in this release in documentation? 

Or maybe:

The tool can be used to deploy, benchmark and compare dynamo deployments on a Kubernetes cluster as well as benchmark and compare servers deployed separately given a URL. In the examples below dynamo deployments are specified with a yaml and servers deployed separately by URL. 

As this quickly can become confusing - I will defer to @itay

Reply:
I'm fine with the updated copy that @nnshah1 proposed:

```
The tool can be used to deploy, benchmark and compare dynamo deployments on a Kubernetes cluster as well as benchmark and compare servers deployed separately given a URL. In the examples below dynamo deployments are specified with a yaml and servers deployed separately by URL.
```

But also fine if we resolve this after. Defer to @hhzhang16

Reply:
I like that wording, I'll use something like that!